lib/puppet/parser/functions/is_domain_name.rb

```diff
@@ -32,6 +32,7 @@ module Puppet::Parser::Functions
     # Check the whole domain
     return false if domain.empty?
     return false if domain.length > domain_max_length
+    return false if /^\d+\.\d+\.\d+\.\d+$/.match(domain)
```

Comment:
I agree that the match should only consider the highest-order label (aka top-level domain) because digits are not permitted there, which would also excluded dotted-quads. Using IPv4 or IPv6 tests would be too strict, I think, because there are dotted quads that would be invalid according to Resolv::IPv4::Regex (such as '192.168.1.500') but should still fail is_domain_name().